### PR TITLE
fix: insert wallet RLS on `auth_user` and updates to `create_new_account`/`addr_addr_to_account`

### PR DESCRIPTION
**Pull request by blushi**
## Description

ref: regen-network/regen-registry#1503

Proposed solution after discussing this with @wgwz (https://github.com/regen-network/regen-web/pull/1918#issuecomment-1578137773):
- only allow `auth_user`s to insert new wallet
- fix existing `create_new_account` fn to handle the case where there's already a party and wallet with an address but no account (created through the "add new profile" in the Roles form) so that later on a user can claim this address and log in with it while keeping the existing profile data
- fix `addr_addr_to_account` fn to not update party_type in case the party already exists
- fix `get_parties_by_name_or_addr` to use left join so that parties without any wallet address are returned from the query (ref: https://github.com/regen-network/regen-web/pull/1918#issuecomment-1580574935)

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] targeted `dev` branch
- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [ ] submit a PR against `master` branch after this one (and other PRs depending on this one, e.g. on regen-network/regen-web, if applicable) get merged

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: migrations/committed/000006.sql [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            addr = v_addr
+        RETURNING
+            id INTO v_wallet_id;
+
+        INSERT INTO party (account_id, TYPE, wallet_id)
+            VALUES (v_account_id, v_party_type, v_wallet_id)
+        ON CONFLICT ON CONSTRAINT
+            party_wallet_id_key
+        DO UPDATE SET
+            account_id = v_account_id
```
Comment (blushi) [inline migrations/committed/000006.sql:41]: this is the main change in this function

File: migrations/committed/000006.sql [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+        IF v_party_id is null THEN
+          RAISE LOG 'creating new party...';
+          SELECT * INTO v_party_id from uuid_generate_v1();
+          INSERT INTO party (id, account_id, TYPE, wallet_id)
+               VALUES (v_party_id, v_account_id, v_party_type, v_wallet_id);
+          RAISE LOG 'new _party_id %', v_party_id;
+        ELSE
+          RAISE LOG 'associating preexisting party...';
+          UPDATE party SET account_id = v_account_id WHERE id = v_party_id ;
```
Comment (blushi) [inline migrations/committed/000006.sql:96]: this is the main change in this function, just updating account_id instead both account_id and party_type

File: server/__tests__/db/functions/authn/add_addr_to_account.test.ts [hunk 3]
```diff
@@ -39,4 +44,40 @@ describe('add_addr_to_account', () => {
       expect(addrs.rowCount).toBe(2);
     });
   });
+  it('allows adding an existing addr with no account yet', async () => {
```
Comment (wgwz) [inline server/__tests__/db/functions/authn/add_addr_to_account.test.ts:47]: thanks for adding this test!

File: server/__tests__/db/functions/authn/create_new_account.test.ts [hunk 4]
```diff
@@ -22,4 +22,35 @@ describe('create_new_account', () => {
       ).rejects.toThrow('this addr belongs to a different account');
     });
   });
+  it('should be able to create a new account with an existing wallet address and party with no account yet', async () => {
```
Comment (wgwz) [inline server/__tests__/db/functions/authn/create_new_account.test.ts:25]: great!

File: server/__tests__/db/functions/get_parties_by_name_or_addr.test.ts [hunk 5]
```diff
@@ -33,4 +32,21 @@ describe('get_parties_by_name_or_addr', () => {
       expect(partiesByNameRes.rows[0].id).toEqual(partyId);
     });
   });
+
+  it('returns parties without wallet addresses by name', async () => {
```
Comment (wgwz) [inline server/__tests__/db/functions/get_parties_by_name_or_addr.test.ts:36]: 👍

**Review by wgwz — APPROVED** [review]
LGTM! thanks for the tests, made it very easy to review

**Pull request merged**